x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/parser/ExpressionBuilder.java

```diff
@@ -263,6 +263,9 @@ public LikePattern visitPattern(PatternContext ctx) {
         }
 
         String pattern = string(ctx.value);
+        if (pattern == null) {
```

Comment:
Since the pattern always has to be non-null, I think it would be better to report the error at the parser level instead of letting it propagate all the way to the tree which would be invalid anyway...
Just like w do with `pos` below by throwing a `ParsingException`

Reply:
I thought about it and I kind of liked more the separation of Parsing to an illegalarg case.
In my mind null is acceptable in parsing but not for the evaluation of the exception.
But if you prefer the ParsingException I can also do.

@astefan what do you think?

Reply:
Where possible, the arguments should be checked as soon as possible - the whole method checks the validity of the pattern right after parsing - checking its existence is one that was missed.
I see no advantages of postponing the check.